backend/pick-git/src/test/java/com/woowacourse/pickgit/authentication/application/OAuthServiceMockTest.java

```diff
@@ -0,0 +1,176 @@
+        String jwtToken = "jwt token";
+
+        OAuthProfileResponse githubProfileResponse = new OAuthProfileResponse();
+        githubProfileResponse.setName("test");
+        githubProfileResponse.setDescription("hi~");
+
+        User user = new User(
+            githubProfileResponse.toBasicProfile(),
+            githubProfileResponse.toGithubProfile()
+        );
+
+        // mock
+        when(oAuthClient.getAccessToken(code)).thenReturn(oauthAccessToken);
+        when(oAuthClient.getGithubProfile(oauthAccessToken)).thenReturn(githubProfileResponse);
+        when(userRepository.findByBasicProfile_Name(githubProfileResponse.getName())).thenReturn(
+            Optional.of(user));
+        when(jwtTokenProvider.createToken(githubProfileResponse.getName())).thenReturn(jwtToken);
+
+        // when
+        String token = oAuthService.createToken(code);
+
+        // then
+        assertThat(token).isEqualTo(jwtToken);
+        verify(userRepository, times(1)).findByBasicProfile_Name(githubProfileResponse.getName());
+        verify(userRepository, times(1)).save(user);
+        verify(jwtTokenProvider, times(1)).createToken(githubProfileResponse.getName());
+        verify(oAuthAccessTokenDao, times(1)).insert(jwtToken, oauthAccessToken);
+    }
+
+    @DisplayName("JWT 토큰을 통해 AccessTokenDB에서 LoginUser에 대한 정보를 가져온다.")
+    @Test
+    void findRequestUserByToken_ValidToken_ReturnAppUser() {
+        // given
+        String token = "jwt token";
+        String accessToken = "oauth access token";
+        String username = "pick-git";
+
+        // mock
+        when(jwtTokenProvider.getPayloadByKey(token, "username")).thenReturn(username);
+        when(oAuthAccessTokenDao.findByKeyToken(token)).thenReturn(Optional.ofNullable(accessToken));
+
+        // when
+        AppUser appUser = oAuthService.findRequestUserByToken(token);
+
+        // then
+        assertThat(appUser).isInstanceOf(LoginUser.class);
+        assertThat(appUser.getUsername()).isEqualTo(username);
+        assertThat(appUser.getAccessToken()).isEqualTo(accessToken);
+    }
+
+    @DisplayName("AccessTokenDB에 저장되어 있지 않은 JWT 토큰이라면 예외가 발생한다.")
+    @Test
+    void findRequestUserByToken_NotFoundToken_ThrowException() {
+        // given
+        String token = "never saved jwt token";
+        String username = "pick-git";
+
+        // mock
+        when(jwtTokenProvider.getPayloadByKey(token, "username")).thenReturn(username);
+        when(oAuthAccessTokenDao.findByKeyToken(token)).thenReturn(Optional.empty());
+
+        // then
+        assertThatThrownBy(() -> oAuthService.findRequestUserByToken(token))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @DisplayName("빈 JWT 토큰이면 GuestUser를 반환한다.")
+    @Test
+    void findRequestUserByToken_EmptyToken_ReturnGuest() {
+        // when
+        AppUser appUser = oAuthService.findRequestUserByToken(null);
+
+        // then
+        assertThat(appUser).isInstanceOf(GuestUser.class);
```

Comment:
```suggestion
        // when
        AppUser appUser = oAuthService.findRequestUserByToken(null);

        // then
        assertThat(appUser.isGuest()).isTrue();
```

`isInstanceOf()` 대신 다형성을 활용해보는 건 어떨까요?

Reply:
오~ 굳굳!! 수정할게유!